app/assets/javascripts/Components/Result/file_viewer.jsx

```diff
@@ -145,17 +145,27 @@ export class FileViewer extends React.Component {
             this.setFileUrl();
           });
         } else {
-          $.ajax({
-            url: this.props.selectedFileURL,
-            data: {preview: true, force_text: force_text},
+          const requestData = {preview: true, force_text: force_text};
+          const url = this.props.selectedFileURL;
+          const queryString = Object.keys(requestData)
```

Comment:
Okay, so following up with my above comment, in this case because the URL isn't generated using js-routes, we do need to manually encode the query parameters to build the full URL. In these cases, you should use the newer [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) (and the `.toString` method) to do the encoding for you.